pages/en/developer/subgraph-debug-forking.mdx

```diff
@@ -0,0 +1,98 @@
+
+## Ok, what is it?
+
+**Subgraph forking** is the process of lazily fetching entities from *another* subgraph's store (usually a remote one).
+
+In the context of debugging, **subgraph forking** allows you to debug your failed subgraph at block *X* without needing to wait to sync-up to block *X*.
+
+## What?! How?
+
+When you deploy a subgraph to a remote Graph node for indexing and it fails at block *X*, the good news is that the Graph node will still serve GraphQL queries using its store, which is synced-up to block *X*. That's great! This means we can take advantage of this "up-to-date" store to fix the bugs arising when indexing block *X*.
+
+In a nutshell, we are going to *fork the failing subgraph* from a remote Graph node that is guaranteed to have the subgraph indexed up to block *X* in order to provide the locally deployed subgraph being debugged at block *X* an up-to-date view of the indexing state.
+
+## Please, show me some code!
+
+To stay focused on subgraph debugging, let's keep things simple and run along with the [example-subgraph](https://github.com/graphprotocol/example-subgraph) indexing the Ethereum Gravity smart contract.
+
+Here are the handlers defined for indexing `Gravatar`s, with no bugs whatsoever:
+
+```tsx
+export function handleNewGravatar(event: NewGravatar): void {
+  let gravatar = new Gravatar(event.params.id.toHex().toString())
+  gravatar.owner = event.params.owner
+  gravatar.displayName = event.params.displayName
+  gravatar.imageUrl = event.params.imageUrl
+  gravatar.save()
+}
+
+export function handleUpdatedGravatar(event: UpdatedGravatar): void {
+  let gravatar = Gravatar.load(event.params.id.toI32().toString())
+  if (gravatar == null) {
+    log.critical('Gravatar not found!', [])
+    return
+  }
+  gravatar.owner = event.params.owner
+  gravatar.displayName = event.params.displayName
+  gravatar.imageUrl = event.params.imageUrl
+  gravatar.save()
+}
+```
+
+Oops, how unfortunate, when I deploy my perfect looking subgraph to the [HostedService](https://thegraph.com/hosted-service/) it fails with the *"Gravatar not found!"* error.
+
+The usual way to attempt a fix is:
+
+  1. Make a change in the mappings source, which you believe will solve the issue (while I know it won't).
+  2. Re-deploy the subgraph to the [HostedService](https://thegraph.com/hosted-service/) (or another remote Graph node).
+  3. Wait for it to sync-up.
+  4. If it breaks again go back to 1, otherwise: Hooray!
+
+It is indeed pretty familiar to an ordinary debug process, but there is one step that horribly slows down the process: *3. Wait for it to sync-up.*
+
+Using **subgraph forking** we can essentially eliminate this step. Here is how it looks:
+
+  0. Spin-up a local Graph node with the ***appropriate fork-base*** set.
```

Comment:
Should this list and the one on line 83 start with 0?

I like how computers index things, but I guess humans are weird... 😆

Reply:
The idea is that these steps starting with 0 are really "setup" steps and not part of the debug loop.

Yeah... Humans have their quirks, hah.